security/entity_provider.rst

```diff
@@ -169,6 +169,12 @@ forces the class to have the five following methods:
 
 To learn more about each of these, see :class:`Symfony\\Component\\Security\\Core\\User\\UserInterface`.
 
+.. caution::
+
+    Do not actually implement ``eraseCredentials`` when you load your users directly
```

Comment:
In addition to being "dangerous", this method is very confusing, specially for newcomers. Should we deprecate it from the User interface?  Or maybe we should wait for #21068, which could remove the UserInterface.